python/pyarrow/_dataset_parquet.pyx

```diff
@@ -729,6 +825,14 @@ cdef class ParquetFragmentScanOptions(FragmentScanOptions):
             other.thrift_container_size_limit)
         return attrs == other_attrs
 
+    def SetParquetDecryptionConfig(self, ParquetDecryptionConfig config):
+        cdef shared_ptr[CParquetDecryptionConfig] c_config
+        if not isinstance(config, ParquetDecryptionConfig):
+            raise ValueError("config must be a ParquetDecryptionConfig")
+        self._parquet_decryption_config = config
+        c_config = config.unwrap()
+        self.parquet_options.parquet_decryption_config = c_config
```

Comment:
Is there a reason to not included this in the `@parquet_decryption_config.setter`? So that property setter can just contain this code, instead of calling `SetParquetDecryptionConfig`.